docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -313,12 +314,22 @@ type LocalObjectTemplate struct {
   - `spec.topology.workers.machineDeployments[i].name` field must be unique within a Cluster
   
 - For object updates:
-  - If `spec.topology.class` is set, it cannot be unset or modified.
+  - If `spec.topology.class` is set, it cannot be unset.
```

Comment:
@fabriziopandini does this cover the use case adequately?

Reply:
I think that's right, so this is a change to enable that behaviour by removing that validation rule and adding additional validation for compatibility to address the consequences of allowing users to change this field.

Reply:
We should prevent also unmanaged --> managed; what about simply stating that we prevent changing from managed to unmanaged and vice versa?